src/VirtualDOM.purs

```diff
@@ -14,29 +15,21 @@ import VirtualDOM.VTree
 -- operation.  See virtual-dom/docs.jsig for details.                             
 foreign import data PatchObject :: *
 
-foreign import showPatchObjectImpl
-  "var showPatchObjectImpl = JSON.stringify;" :: PatchObject -> String
+foreign import showPatchObjectImpl :: PatchObject -> String
 
 instance showPatchObject :: Show PatchObject where
   show = showPatchObjectImpl
 
-foreign import createElement
-  "var createElement = require('virtual-dom/create-element');" :: VTree -> Node
+foreign import createElement :: VTree -> Node
 
-foreign import diff'
-  "var diff$prime = require('virtual-dom/diff');" :: Fn2 VTree VTree PatchObject
+foreign import diff_ :: Fn2 VTree VTree PatchObject
 
 diff :: VTree -> VTree -> PatchObject
-diff = runFn2 diff'
+diff = runFn2 diff_
 
-foreign import patch'
-  "var patch$prime = require('virtual-dom/patch');" :: Fn2 Node PatchObject Node
+foreign import patch_ :: Fn2 Node PatchObject Node
 
 patch :: forall e. PatchObject -> Node -> Eff (dom :: DOM | e) Node
-patch p n = mkEff \_ -> runFn2 patch' n p
+patch p n = mkEff \_ -> runFn2 patch_ n p
```

Comment:
I'm new to PureScript but I imagine this entails:
  * dropping `mkEff`, since it's just an identity
  * thunking the return value of `patch_` already in the JS code so that the PS type `Eff` corresponds to a JS thunk (this would give the type `Node -> PatchObject -> Eff (dom :: DOM | e) Node` to `patch_`, which would be a more honest type)